Automattic/wp-calypso
# Plans: Reduxify Plan Remove component

This pull request addresses https://github.com/Automattic/wp-calypso/issues/2626 by extracting the part that handles the API call from the `<PlanRemove />` component and connecting this component to the Redux state tree. It also fixes a few things and refactors some code in the process:

##### Fixing

* Fixed redirection failing when clicking success notice
* Removed duplicate action types constants
* Fixed error property not reset when fetching site plans
* Added error handling when fetching site plans
 

##### Refactoring

* Renamed action type constants to follow naming convention
* Updated action for clearing site plans to return an action object
* Standardized descriptions of unit tests and updated them to be easier to reason about
* Added a bunch of new unit tests

##### Reduxification

* Added new actions (and unit tests) for canceling a plan free trial
* Reduxified the `<PlanRemove />` component

As a reminder the `<PlanRemove />` component is responsible for displaying a new section with a link at the bottom of the `Plans Overview` page during the grace period:

![screenshot](https://cloud.githubusercontent.com/assets/594356/12267140/f6d1c21a-b946-11e5-8bd1-d7d7014246cd.png)

This link displays a modal that allows users to cancel a free trial:

![screenshot](https://cloud.githubusercontent.com/assets/594356/12267216/59f2eb44-b947-11e5-97aa-b37656bb8f23.png)

Users are redirected on the `Plans` page with a message if the cancelation is successful:

![screenshot](https://cloud.githubusercontent.com/assets/594356/12388305/551ae4f2-bdd2-11e5-9d3f-ed76c36467a2.png)
 
#### Testing instructions

1. Run `git checkout update/reduxify-plan-remove` and start your server
2. Open the [`Plan Overview` page](http://calypso.localhost:3000/plans) with a free trial in grace period
3. Click the `Remove now` link and then click the `Remove Now` button in the modal
4. Check that the free trial was canceled successfully
5. Check that you are redirected to the `Plans` page
6. Check that you can dismiss the success notice and that no error is thrown

#### Reviews
